<launch>
  <let name="urdf_path" value="$(find-pkg-share vw-assign1-2025-parol6-mesh-pkg)/urdf/PAROL6.urdf.xacro"/>
  <let name="rviz_config_path" value="$(find-pkg-share vw-assign1-2025-parol6-mesh-pkg)/rviz/urdf_config.rviz"/>
  <node pkg="robot_state_publisher" exec="robot_state_publisher">
    <param name="robot_description" value="$(command 'xacro $(var urdf_path)')"/>
  </node>
  <node pkg="joint_state_publisher_gui" exec="joint_state_publisher_gui"/>
  <node pkg="rviz2" exec="rviz2" output="screen" args="-d $(var rviz_config_path)"/>
  <node pkg="vw-assign1-2025-parol6-mesh-pkg" exec="vw-assign1-2025-mesh_pub-node">
    <param name="board_location_x" value="0.1"/>
    <param name="board_location_y" value="0.0"/>
    <param name="board_location_z" value="0.0"/>
    <param name="coin_holder_location_x" value="0.30"/>
    <param name="coin_holder_location_y" value="0.08"/>
    <param name="coin_holder_location_z" value="0.0"/>
    <param name="coin_location_x" value="0.34"/>
    <param name="coin_location_y" value="0.08"/>
    <param name="coin_location_z" value="0.0"/>
  </node>
</launch>
